src/coreclr/jit/morph.cpp

```diff
@@ -10869,7 +10869,71 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
             INDEBUG(node->gtDebugFlags |= GTF_DEBUG_NODE_MORPHED);
             return node;
         }
+#if defined(TARGET_XARCH)
+        case NI_SSE_And:
+        case NI_SSE2_And:
+        case NI_AVX_And:
+        case NI_AVX2_And:
+        {
+            if (node->GetOperandCount() != 2)
+            {
+                return node;
+            }
+
+            GenTree*            op1      = node->Op(1);
+            GenTree*            op2      = node->Op(2);
+            GenTree*            lhs      = nullptr;
+            GenTree*            rhs      = nullptr;
+            GenTreeHWIntrinsic* inner_hw = nullptr;
+
+            // Transforms ~v1 & v2 to VectorXxx.AndNot(v2, v1)
+            if (op1->OperIs(GT_HWINTRINSIC))
+            {
+                rhs      = op2;
+                inner_hw = op1->AsHWIntrinsic();
+            }
+            // Transforms v2 & (~v1) to VectorXxx.AndNot(v1, v2)
+            else if (op2->OperIs(GT_HWINTRINSIC))
+            {
+                rhs      = op1;
+                inner_hw = op2->AsHWIntrinsic();
+            }
+            else
+            {
+                return node;
+            }
+
+            if ((inner_hw->GetOperandCount() != 2) || (!inner_hw->Op(2)->IsVectorAllBitsSet()))
+            {
+                return node;
+            }
```

Comment:
Would be better to check this as part of handling `_Xor` below, that way you don't need to check the operand count and its easier for the general logic to support `AdvSimd_Not` on Arm64.